PUDL Data Release v2025.8.0

[Additional release process documentation](https://catalystcoop-pudl.readthedocs.io/en/nightly/dev/run_a_release.html).

# Release Checklist

- [x] Set a release date & notify team
- [x] Update our CITATION.cff file with the new release date and current Catalyst membership.
- [x] Update our .zenodo.json file with current Catalyst membership.
- [x] Close out the [PUDL Release Notes](https://catalystcoop-pudl.readthedocs.io/en/nightly/release_notes.html) with an overview of the changes in this release
- [x] Merge those changes into `main`
- [x] Push v2025.8.0 tag (4-digit year, month with no leading zero, and patch version) to `main`
- [x] Verify that the new [GitHub (software) release](https://github.com/catalyst-cooperative/pudl/releases) has been published
- [x] Verify [`catalystcoop.pudl` PyPI (software) release](https://pypi.org/project/catalystcoop.pudl/)
- [x] Verify that [PUDL repo archive on Zenodo](https://zenodo.org/doi/10.5281/zenodo.3404014) has been updated w/ new version
- [x] Wait 3-5 hours for a successful build to complete
- [x] Activate new version on the [RTD admin panel](https://app.readthedocs.org/projects/catalystcoop-pudl/) and verify that it builds successfully.
- [ ] ~Verify that `stable` and the version tag point at same git ref~ Not possible due to funky post v2025.7.0 commits
- [x] Verify that [`stable` docs on RTD](https://catalystcoop-pudl.readthedocs.io/en/stable/) have been updated
- [x] Verify `gs://pudl.catalyst.coop/v2025.8.0` has the new expected data.
- [x] Verify `gs://pudl.catalyst.coop/stable` has the new expected data.
- [x] Verify `s3://pudl.catalyst.coop/v2025.8.0` has the new expected data.
- [x] Verify `s3://pudl.catalyst.coop/stable` has the new expected data.
- [x] Verify that draft deposition for [this Zenodo archive](https://doi.org/10.5281/zenodo.3653158) has the new expected data
- [x] Update Zenodo description and other metadata based on release notes (see task list below)
- [x] Manually publish the new Zenodo deposition with the updated metadata
- [x] Wait 2-12 hours for a bot to create a PR in the [PUDL conda-forge feedstock](https://github.com/conda-forge/catalystcoop.pudl-feedstock/pulls)
- [x] Review the `conda-forge` PR, updating dependencies and CLI entrypoints if necessary. Our direct dependencies (from `pyproject.toml`) should be pinned to the actual version appearing in `environments/conda-linux-64.lock.yml`
- [x] Create an Announcement for the release in [our GitHub Discussions](https://github.com/orgs/catalyst-cooperative/discussions)
- [ ] Update the [release documentation](https://catalystcoop-pudl.readthedocs.io/en/nightly/dev/run_a_release.html) to better reflect the actual process for next time

## Zenodo Metadata Tasks

- [x] Make sure all Catalyst members are listed in creators (automate!)
- [x] Update Zenodo deposition description to reflect current release (automate!)
- [x] Set correct version for Zenodo data release archive (automate!)
- [x] Set keywords for Zenodo data release archive (automate!)
- [x] Set language of Zenodo data release archive (automate!)
- [x] Zenodo PUDL repo archive: update release notes (cut-and-paste from GitHub)

## Issues that we ran into

### 502 Server Error in GitHub Release

Got a 502 Server Error while trying to create the GitHub release via our `release` workflow. Tried re-running the failed job but got the same error, without a lot of useful information about why. On the 3rd retry it worked.

```
Run gh release create 'v2025.8.0' ./dist/*.tar.gz ./dist/*.whl --title 'PUDL v2025.8.0' --generate-notes --repo 'catalyst-cooperative/pudl'
  gh release create 'v2025.8.0' ./dist/*.tar.gz ./dist/*.whl --title 'PUDL v2025.8.0' --generate-notes --repo 'catalyst-cooperative/pudl'
  shell: /usr/bin/bash -e {0}
  env:
    GHA_SIGSTORE_PYTHON_INTERNAL_SIGNING_ARTIFACTS: /home/runner/work/pudl/pudl/dist/catalystcoop_pudl-2025.8.0.tar.gz
  /home/runner/work/pudl/pudl/dist/catalystcoop_pudl-2025.8.0.tar.gz.sigstore.json
  /home/runner/work/pudl/pudl/dist/catalystcoop_pudl-2025.8.0-py3-none-any.whl
  /home/runner/work/pudl/pudl/dist/catalystcoop_pudl-2025.8.0-py3-none-any.whl.sigstore.json
    GITHUB_TOKEN: ***
HTTP 502: Server Error (https://api.github.com/repos/catalyst-cooperative/pudl/releases)
Error: Process completed with exit code 1.
```

### Changelog still wrong

The automatic detection of previous release is still failing in the generation of the release changelog. Had to be re-run with manually selected previous version.

### Stable branch update failed

Due to "diverging branches" -- but I'm not sure what was different.

```
+ git config --unset http.https://github.com/.extraheader
+ git config user.email pudl@catalyst.coop
+ git config user.name pudlbot
+ set +x
Setting authenticated git remote URL using PAT
+ echo 'Updating stable branch to point at v2025.8.0.'
Updating stable branch to point at v2025.8.0.
+ git checkout -- dbt/seeds/
+ git fetch --force --tags origin v2025.8.0
From https://github.com/catalyst-cooperative/pudl
 * tag                   v2025.8.0  -> FETCH_HEAD
 t [tag update]          v2025.8.0  -> v2025.8.0
+ git fetch origin stable:stable
From https://github.com/catalyst-cooperative/pudl
 * [new branch]          stable     -> stable
+ git checkout stable
Previous HEAD position was a5b47564b Close out release notes for PUDL v2025.8.0 (#4543)
Switched to branch 'stable'
+ git show-ref -d stable v2025.8.0
ba234e0fdb3fbd6b3e30578911e4d27d79be6d22 refs/heads/stable
ba234e0fdb3fbd6b3e30578911e4d27d79be6d22 refs/remotes/origin/stable
7d1d055693bf684155fdcb2a851676fbbd456d5e refs/tags/v2025.8.0
a5b47564bcd58921ea6bf26b29d7475dafa6b830 refs/tags/v2025.8.0^{}
+ git merge --ff-only v2025.8.0
error: cannot run gpg: No such file or directory
hint: Diverging branches can't be fast-forwarded, you need to either:
hint: 
hint: 	git merge --no-ff
hint: 
hint: or:
hint: 
hint: 	git rebase
hint: 
hint: Disable this message with "git config advice.diverging false"
fatal: Not possible to fast-forward, aborting.
```

I get the same error if I attempt to update the branch to point at the tag locally with a clean working tree, so I don't think it's that there's some modified file laying around.  It looks like there are some commits on `stable` that don't exist on `v2025.8.0`? Is this a rebase gone wrong?  What's the safe way to resolve this discrepancy? @jdangerx 

```console
git log v2025.8.0..stable --oneline | cat
ba234e0fd add release note for removing superset
1d62c6b57 Close out release notes for v2025.7.0
81ba32797 Update citation and zenodo metadata for release.
```

The `git merge --no-ff` option gets to where there's no difference between `stable` and `v2025.8.0` but they aren't the same commit, which is annoying.

--- a/.github/workflows/pytest.yml
+++ b/.github/workflows/pytest.yml
@@ -39,6 +39,10 @@ jobs:
               - '!docs/**.rst.jinja'
               - '!docs/**.bib'
               - '!README.rst'
+              - '!CONTRIBUTING.rst'
+              - '!LICENSE.txt'
+              - '!CITATION.cff'
+              - '!.zenodo.json'
       - name: print out components
         run: |
           echo "event name (${{ github.event_name }}) is workflow dispatch: ${{ github.event_name == 'workflow_dispatch' }}"

--- a/CITATION.cff
+++ b/CITATION.cff
@@ -23,6 +23,6 @@ authors:
     given-names: Kathryn
 
 title: "The Public Utility Data Liberation (PUDL) Project"
-version: 2025.7.0
+version: 2025.8.0
 doi: 10.5281/zenodo.3404014
-date-released: 2025-07-03
+date-released: 2025-08-14

--- a/docs/release_notes.rst
+++ b/docs/release_notes.rst
@@ -3,8 +3,21 @@ PUDL Release Notes
 =======================================================================================
 
 ---------------------------------------------------------------------------------------
-v2025.XX.x (2025-MM-DD)
+v202X.XX.x (YYYY-MM-DD)
 ---------------------------------------------------------------------------------------
+
+.. _release-v2025.8.0:
+
+---------------------------------------------------------------------------------------
+v2025.8.0 (2025-08-14)
+---------------------------------------------------------------------------------------
+
+This is a regular quarterly release of PUDL. It includes new 2024 annual updates for a
+number of datasets (FERC Forms 2, 6, 60, & 714), and a minor update to the 2024 FERC
+Form 1 data that includes late filings & revisions. It also includes year-to-date
+updates for the monthly and quarterly datasets, including EIA-860M, EIA-923, EIA-930,
+and the EPA CEMS hourly emissions. There were also a number of data processing bug fixes
+and data usability improvements. See the full notes below for details.
 
 New Data
 ^^^^^^^^
@@ -24,6 +37,11 @@ EIA-860M
 * Updated EIA-860M monthly generator report with newly published data for May and June
   of 2025. See issue :issue:`4379` and PR :pr:`4536`.
 
+EIA-923
+~~~~~~~
+
+* Added EIA-923 data through May 2025. See :issue:`4516` and :pr:`4538`.
+
 EIA 930
 ~~~~~~~
 
@@ -35,6 +53,11 @@ EIA Bulk Electricity API
 
 * Updated the EIA Bulk Electricity data to include data published up through
   the beginning of August 2025. See :issue:`4519` and PR :pr:`4523`.
+
+EPA CEMS
+~~~~~~~~
+
+* Added EPA CEMS data through June 2025. See :issue:`4518` and :pr:`4531`.
 
 FERC Form 1
 ~~~~~~~~~~~
@@ -48,12 +71,10 @@ FERC Forms 2, 6 and 60
 * Updated our extraction of FERC Forms 2, 6, and 60 to raw SQLite databases to include
   2024 data. See :issue:`4418` and :pr:`4433`.
 
-FERC Forms 714
-~~~~~~~~~~~~~~
+FERC Form 714
+~~~~~~~~~~~~~
 
 * Integrated 2024 data for FERC Form 714. See issue :issue:`4409` and PR :pr:`4530`.
-* Updated FERC Form 1 2024 data to include late respondents. See :issue:`4493` and
-  :pr:`4507`.
 
 PHMSA Gas Data
 ~~~~~~~~~~~~~~
@@ -63,14 +84,6 @@ PHMSA Gas Data
   :pr:`4470`.
 * Extracted 1970 through 1989 PHMSA transmission data to raw assets.  This data is not
   currently published to the PUDL database. See :issue:`3290` and :pr:`4500`.
-
-EPACEMS
-~~~~~~~
-* Added EPACEMS data through June 2025. See :issue:`4518` and :pr:`4531`.
-
-EIA-923
-~~~~~~~
-* Added EIA-923 data through May 2025. See :issue:`4516` and :pr:`4538`.
 
 Quality of Life Improvements
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

